### PR TITLE
Added EXR support for syslog_server

### PR DESCRIPTION
**Pull request by jonnytdevops**

### CODE REVIEW

File: lib/cisco_node_utils/syslog_server.rb [hunk 1]
```diff
@@ -62,25 +82,64 @@ def ==(other)
...
-                 ip:    "#{name}",
-                 level: level.nil? ? '' : "#{level}",
-                 vrf:   vrf.nil? ? '' : "use-vrf #{vrf}",
-                )
+      if platform == :ios_xr
+
+        # This provider only support a 1-1 mapping between host and VRF.
+        # Thus, we must remove the other entries on different VRFs.
+        all_vrfs = config_get('syslog_server', 'vrf', name)
+        destroy(all_vrfs) if all_vrfs.count > 1
```
Comment (glennmatthews) [inline lib/cisco_node_utils/syslog_server.rb]:
```
ArgumentError: wrong number of arguments (0 for 1+)
    lib/cisco_node_utils/syslog_server.rb:90:in `count'
```

Comment (jonnytdevops) [inline lib/cisco_node_utils/syslog_server.rb]: Strange..I don't get this error. Could this be a ruby version thing? I'm confused. Maybe it's something obvious I'm missing?

Comment (glennmatthews) [inline lib/cisco_node_utils/syslog_server.rb]: Are you testing against XR? Can you provide minitest results?

Comment (jonnytdevops) [inline lib/cisco_node_utils/syslog_server.rb]: I'm verifying manually on XR, as we don't have the minitests working yet due to a image deployment issue. I've opened a ticket for this with the relevant team.

Comment (jonnytdevops) [inline lib/cisco_node_utils/syslog_server.rb]: I managed to reproduce this error. Fixing.... :)

**Comment by jonnytdevops** [comment]
@glennmatthews Code updated, thanks!

File: tests/test_syslog_server.rb [hunk 2]
```diff
@@ -34,9 +34,15 @@ def teardown
 
   def no_syslogserver
     # Turn the feature off for a clean test.
-    config('no logging server 1.2.3.4',
-           'no logging server 2003::2',
-           'no vrf context red')
+    if platform == :ios_xr
+      config('no logging 1.2.3.4',
+             'no logging 2003::2',
+             'no vrf red')
```
Comment (glennmatthews) [inline tests/test_syslog_server.rb:42]: Seems like a bug in IOS XR to me, but 'no vrf red' doesn't remove 'logging 2003::2 vrf red'. I had to make the following addition to get minitest passing for me:

``` diff
diff --git a/tests/test_syslog_server.rb b/tests/test_syslog_server.rb
index 520f775..772bc74 100644
--- a/tests/test_syslog_server.rb
+++ b/tests/test_syslog_server.rb
@@ -36,7 +36,9 @@ class TestSyslogServer < CiscoTestCase
     # Turn the feature off for a clean test.
     if platform == :ios_xr
       config('no logging 1.2.3.4',
+             'no logging 1.2.3.4 vrf red',
              'no logging 2003::2',
+             'no logging 2003::2 vrf red',
              'no vrf red')
     else
       config('no logging server 1.2.3.4',
```

**Comment by jonnytdevops** [comment]
@glennmatthews thanks glenn, I've re-pushed the code with your 2 lines added. Thanks!

**Comment by glennmatthews** [comment]
:+1:

@mikewiebe @chrisvanheuveln @robert-w-gries Can I get a :+1: from the Nexus side?

**Comment by mikewiebe** [comment]
:+1:

**Pull request merged**